sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -811,17 +811,19 @@ public final Double getFractionConsumed() {
     public final synchronized BigtableSource splitAtFraction(double fraction) {
       ByteKey splitKey;
       try {
-        splitKey = source.getRange().interpolateKey(fraction);
+        splitKey = rangeTracker.getRange().interpolateKey(fraction);
       } catch (IllegalArgumentException e) {
-        logger.info("%s: Failed to interpolate key for fraction %s.", source.getRange(), fraction);
+        logger.info(
+            "%s: Failed to interpolate key for fraction %s.", rangeTracker.getRange(), fraction);
         return null;
       }
       logger.debug(
           "Proposing to split {} at fraction {} (key {})", rangeTracker, fraction, splitKey);
       if (!rangeTracker.trySplitAtPosition(splitKey)) {
         return null;
       }
-      BigtableSource primary = source.withEndKey(splitKey);
+      BigtableSource primary =
```

Comment:
This is subtle, but I don't think that we want this change. If the Source's range is `[a,z)` and the first key is `d`, I think the source's range is still `[a, z)`. Leave the start key alone, but the rangeTracker will still keep the state the same.

(The change above is still good)